runtime/lib/cloudapi/index.js

```diff
@@ -140,21 +150,29 @@ CloudStore.prototype.handleResponse = function handleResponse (data) {
  * @method
  */
 CloudStore.prototype.syncDate = function syncDate () {
-  fs.readFile('/tmp/LOGIN_HEADER', 'utf8', (err, text) => {
-    if (err || !text) {
-      logger.warn('/tmp/LOGIN_HEADER invalid body, discard sync')
-      return
-    }
+  return readFileAsync('/tmp/LOGIN_HEADER', 'utf8').then((text) => {
     try {
       var headers = parseHttpHeader(text)
+      var error
       if (headers && headers.date) {
-        sync(headers.date)
+        return sync(headers.date)
       } else {
         logger.warn('skip sync date, reason: no date found from headers', text)
+        error = new Error('skip sync date, reason: no date found from headers')
+        error.code = 'SYNC_DATE'
+        throw error
       }
     } catch (err) {
       logger.warn(`sync date error: ${err && err.message}`)
+      error = new Error('skip sync date, reason: no date found from headers')
+      error.code = 'SYNC_DATE'
```

Comment:
Duplicated error handling if the error is thrown at https://github.com/yodaos-project/yodart/pull/120/files#diff-271915d1c6641f71b9bd287ff3d51f7fR163